src/core/libmaven/jsonReports.cpp

```diff
@@ -1,286 +1,678 @@
-                                mavenParameters->filterline);
-        }
-
-        //TODO: for MS1 we've already limited RT range, but for MS/MS the entire RT range of the SRM will be output
-        //either check here or edit getEIC functionality
-        if(eic) {
-            int N = eic->rt.size();
-
-            //myfile << setprecision(4);
-
-            myfile << ",\n" << "\"eic\": {" ;
-
-            myfile << "\"rt\": [";
-
-            for(int i=0;i<N;i++){
-                if ( eic->rt[i] > 0) {
-                    myfile << eic->rt[i];
-                    if (i < N - 1) myfile << ",";
-                }
-            }
-            myfile << "],\n" ; //rt
-            myfile << "\"intensity\": [";
-            for(int i=0;i<N;i++){
-                if ( eic->rt[i] > 0) {
-                    myfile << eic->intensity[i];
-                    if (i < N - 1) myfile << ",";
-                }
-            }
-            myfile << "]" ; //intensity
-            myfile << "\n}" ;//eic
-
-            myfile << "\n}" ; //peak
-            delete(eic);
-        }
-    }
-
-
-    myfile << "\n]" ; //peaks
-
-    myfile << "}" ; //group
-}
-
-
-void JSONReports::saveMzEICJson(string filename,vector<PeakGroup> allgroups,vector<mzSample*> samples) {
-    ofstream myfile(filename.c_str());
-    myfile << setprecision(10);
-
-    myfile << "{\"groups\": [" <<endl;
-
-    int groupId=0;
-    int metaGroupId=0;
-    vector<mzSample*> vsamples = samples;
-
-    for(int i=0; i < allgroups.size(); i++ ) {
-        PeakGroup& grp = allgroups[i];
-
-        //if compound is unknown, output only the unlabeled form information
-        if( grp.getCompound() == NULL || grp.childCount() == 0 ) {
-            grp.groupId= ++groupId;
-            grp.metaGroupId= ++metaGroupId;
-            if(groupId>1) myfile << "\n,";
-            writeGroupMzEICJson(grp,myfile, vsamples);
-        }
-        else { //output all relevant isotope info otherwise
-            //does this work? is children[0] always the same as grp (parent)?
-            grp.metaGroupId = ++ metaGroupId;
-            for (unsigned int k=0; k < grp.children.size(); k++) {
-                grp.children[k].metaGroupId = grp.metaGroupId;
-                grp.children[k].groupId= ++groupId;
-                if(groupId>1) myfile << "\n,";
-                writeGroupMzEICJson(grp.children[k], myfile, vsamples);
-            }
-        }
-        //Q_EMIT(updateProgressBar("Writing to json file. Please wait...", i, allgroups.size() - 1));
-    }
-    myfile << "]}"; //groups
-    myfile.close();
-    //Q_EMIT(updateProgressBar("Writing to json complete.", 1, 1));
-}
-
-string JSONReports::sanitizeJSONstring(string s) {
-    boost::replace_all(s, "\"","\\\"");
-    //replace(s.begin(),s.end(),"\"","\\\""); //escape quotes
-    s="\""+s+"\""; //quote the whole string
-    return s;
-}
+#include "doctest.h"
+#include "json.hpp"
+
+using json = nlohmann::json;

```

Comment:
Let us cluster includes before any namespace declarations. And in general, follow the order:
1. Defines (#define) go first, then includes (#include) and then other preprocessor macros.
2. Namespace, constants, forward declarations, etc. will follow preprocessor directives.